libs/dal/src/types/helpers.ts

```diff
@@ -1,5 +1,119 @@
 import { Types } from 'mongoose';
 
-export type ChangePropsValueType<T, K extends keyof T, V = Types.ObjectId> = Omit<T, K> & {
-  [P in K]: V;
+/**
+ * The type of an ObjectKey identifier.
+ * This is the type of the `_id` field in a Mongoose document
+ *
+ * This is intentionally type widened to `string` to cater for
+ * cases where the `_id` field is a string instead of a `Types.ObjectId`
+ * TODO: Investigate and decide on approach to choose either `Types.ObjectId` or `string`
+ *
+ * @see https://mongoosejs.com/docs/schematypes.html#objectids
+ */
+export type ObjectIdType = Types.ObjectId | string;
+
+/**
+ * The type union of the PK identifier and the template literal type of an FK identifier
+ */
+export type ObjectIdKey = '_id' | `_${string}Id`;
```

Comment:
Cute! I was not aware of the benefit of using Template Literal in Types like this.